TravelAgency-service/src/test/java/cz/muni/fi/travelAgency/facade/TripFacadeTest.java

```diff
@@ -0,0 +1,129 @@
+
+    @Spy
+    @Inject
+    private BeanMappingService beanMappingService;
+
+    @BeforeClass
+    public void InitClass() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @BeforeMethod
+    public void SetupTripDTO(){
+        tripDTO = new TripDTO();
+        tripDTO.setId(1L);
+        tripDTO.setFromDate(firstDate);
+        tripDTO.setToDate(secondDate);
+        tripDTO.setDestination(destination);
+        tripDTO.setCapacity(10);
+        tripDTO.setPrice(100.20);
+    }
+
+    @Test
+    public void createTripTest(){
+        tripFacade.createTrip(tripDTO);
+        tripBrno = beanMappingService.mapTo(tripDTO, Trip.class);
+        Assert.assertEquals(tripBrno.getDestination(),destination);
+        Assert.assertEquals(tripBrno.getCapacity(),10);
+        Mockito.verify(tripService).createTrip(tripBrno);
+    }
+
+    @Test
+    public void updateTripTest() {
+        tripDTO.setFromDate(firstDate);
+        tripDTO.setToDate(secondDate);
+        tripDTO.setDestination("New Lake Island");
+        tripDTO.setCapacity(15);
+        tripDTO.setPrice(150.20);
+        tripFacade.updateTrip(tripDTO);
+        tripBrno = beanMappingService.mapTo(tripDTO, Trip.class);
+        Assert.assertEquals(tripBrno.getDestination(),"New Lake Island");
+        Assert.assertEquals(tripBrno.getCapacity(),15);
+        Mockito.verify(tripService).updateTrip(tripBrno);
+    }
+
+    @Test
+    public void removeTripTest() {
+        Assert.assertNotNull(tripDTO);
+        tripFacade.removeTrip(tripDTO);
+        tripBrno = beanMappingService.mapTo(tripDTO, Trip.class);
+        Mockito.verify(tripService).removeTrip(tripBrno);
+    }
+
+    @Test
+    public void testFindById(){
+        tripBrno = beanMappingService.mapTo(tripDTO, Trip.class);
+        Mockito.when(tripService.findById(tripDTO.getId())).thenReturn(tripBrno);
+        TripDTO tripFromFacade = tripFacade.getTripById(tripDTO.getId());
+        Assert.assertEquals(tripFromFacade.getDestination(),destination);
+    }
+
+    @Test
+    public void getAvailableSlotsTripTest() {
+        Trip trip1 = new Trip();
+        trip1.setId(11L);
+        trip1.setFromDate(firstDate);
+        trip1.setToDate(secondDate);
+        trip1.setDestination(destination);
+        trip1.setCapacity(1);
+        trip1.setPrice(100.20);
+
+        Trip trip2 = new Trip();
+        trip2.setId(12L);
+        trip2.setFromDate(firstDate);
+        trip2.setToDate(secondDate);
+        trip2.setDestination("Prague");
+        trip2.setCapacity(6);
+        trip2.setPrice(120.20);
+
+        Trip trip3 = new Trip();
+        trip3.setId(13L);
+        trip3.setFromDate(firstDate);
+        trip3.setToDate(secondDate);
+        trip3.setDestination("Paris");
+        trip3.setCapacity(8);
+        trip3.setPrice(320.20);
+        List<Trip> allTrips = Arrays.asList(trip1,trip2,trip3);
+
+        Mockito.when(tripService.findTripBySlot(2)).thenReturn(allTrips);
```

Comment:
Why return all trips?? You are testing the search for capacit  == 2  this test does not make sense, please rewrite.